### ISSUE TITLE
Add selector-pseudo-element-no-invalid rule

### ISSUE THREAD

**Issue by salzhrani** [stylelint/stylelint#2155]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

a new rule warn about writing a decedent to a pseudo element

> Which rule, if any, is this issue related to?

```css
.foo::after .bar {
  color: pink;
}
```

> Which version of stylelint are you using?

7.6.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

e.g. "CLI with `stylelint --config myconfig *.css`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

it also affects scss

```scss
.foo::after {
  .bar {
    color: pink;
  }
}
```

> What did you expect to happen?

a warning like: Pseudo elements have no descendants

> What actually happened (e.g. what warnings or errors you are getting)?

no errors

**Comment by jeddy3** [comment]
@salzhrani Thank you for the suggestion and for using the template.

@stylelint/core We can discuss this in https://github.com/stylelint/stylelint/issues/2126 as its another spec validation rule.

**Comment by evilebottnawi** [comment]
@jeddy3 css validation have error, I think this does not apply to us, best solution is plugin

**Comment by davidtheclark** [comment]
If we do make a rule for this, I think it should be a little more general, like `selector-pseudo-element-no-invalid`. [In the spec](https://www.w3.org/TR/selectors4/#pseudo-classes) there are some specific rules:

> - only one pseudo-element may appear per complex selector 
> - the pseudo-element must appear after the compound selector that represents the subjects of the selector
> - the pseudo-element may appear only if the subject of the selector is the last compound selector in the selector.

**Comment by jeddy3** [comment]
> If we do make a rule for this, I think it should be a little more general, like selector-pseudo-element-no-invalid. 

Agreed that that would be best regardless of whether this is a plugin or a core rule. 

@salzhrani FYI, we're currently [doing an audit](https://github.com/stylelint/stylelint/issues/2079) of the existing rules and formalising the boundary between core rules and plugins. Your suggestion falls close to where the boundary might fall, as such there might be a wait on resolving this issue. As an aside, if we do split spec enforcement rules into a plugin pack, would you interested in championing such a thing and helping to maintain it?

**Comment by davidtheclark** [comment]
I think that there should be an independent tool that validates selectors. Or maybe this could be an addition to `postcss-selector-parser`. Then we could have a single `selector-no-invalid` rule. 

Looks like csstree has selector validation [on the roadmap](https://github.com/csstree/csstree/issues/34). Somebody could write a standalone package for this, too.

**Comment by jeddy3** [comment]
> Looks like csstree has selector validation on the roadmap.

Yep, it looks like there are some exciting things happening over at csstree when it comes to spec validation!

**Comment by evilebottnawi** [comment]
@davidtheclark @jeddy3 agree, closing. If in the future we decide to validate rules in the core, then this is a separate issue :+1:
